### ISSUE TITLE
Request for a Mac version

### ISSUE THREAD

**Issue by BankofSwax** [Franiac/TwitchLeecher#190]
Hey guys thanks for the hard work, but I wonder if you guys could release a mac os version too? It would be very helpful to me as I constantly working on a mac, thank you and have a nice day.

**Comment by Franiac** [comment]
Since TL is implemented in C# for.NET 4.5 there is no option at the moment to get it onto the Mac. I plan to rewrite the whole program in HTML 5 / JS / CSS. This way I will be able to bring it to Windows, Linux, MacOS, iOS, Android and Windows UWP via Electron and Cordova with a single code basis.

The big problem is that I have to put the "download and processing part" into the cloud (Azure, S3 or Google) because ffmpeg is not available on mobile devices. But the cloud server farm is way too expensive for me (couple of thousands of $ per month).

So basically I could do all that but since TL is open source and generates no income it is very unlikely that this all will happen anytime soon.

**Comment by entdark** [comment]
There is ffmpeg for Android though. I am using it.

**Comment by Franiac** [comment]
There is a compile for that, true, but that does not help if it is not running on UWP and iOS also. I will not start to implement platform specific versions of TL with different code bases.

**Comment by garoto** [comment]
There's always [youtube-dl](https://github.com/rg3/youtube-dl/)

**Comment by teocci** [comment]
https://github.com/teocci/TwitchLeecher I created this project for the people that uses mac or linux.

**Comment by rosshadden** [comment]
@garoto I forgot to thank you for the suggestion. youtube-dl works amazingly. I have been using it for months based on your recommendation.
